baseapp/baseapp.go

```diff
@@ -568,6 +568,12 @@ func (app *BaseApp) getState(mode runTxMode) *state {
 	return app.checkState
 }
 
+// NewProcessProposalContext returns a context with a branched version of the
+// DeliverTx state that is safe to use during ProcessProposal.
+func (app *BaseApp) NewProcessProposalContext(h tmproto.Header) sdk.Context {
+	return sdk.NewContext(app.deliverState.ms.CacheMultiStore(), h, false, app.logger)
```

Comment:
the main diff from existing functionality is that we are using a branch of the deliver state, since we want to be sure that do not change anything and we want to be sure that we're using the deliver state. We could probably get away with just using the deliver state since we're currently only reading, or simply cache the `ms` instead of the deliver state, but this makes the most sense to me atm.